### PR TITLE
refactor: split coach summary ownership and settings modules

### PR DESCRIPTION
**Pull request by andrzejwitkowski**
## Summary
- split AI Coach summary ownership into focused cache, socket, summary-state, and workout-list helpers without changing user-visible behavior
- split backend settings service and Mongo settings repository into concern-based modules and split their tests accordingly
- add regression coverage for stale metadata hydration, websocket reconnect stability, and refreshed omitted summaries

## Verification
- bun run --cwd frontend test src/features/coach/context/CoachSessionCache.test.tsx src/features/coach/hooks/useCoachChat.connection.test.tsx src/features/coach/hooks/useCoachChat.sendMessage.test.tsx src/features/coach/hooks/useCoachChat.saveWorkflow.test.tsx src/features/coach/hooks/useCoachChat.staleSelection.test.tsx src/features/coach/hooks/useWorkoutList.test.tsx src/features/coach/components/CoachPageLayout.test.tsx
- bun run verify:arch
- rtk cargo fmt --all --check
- rtk cargo clippy --all-targets --all-features -- -D warnings

## Notes
- graphify rebuild is still blocked in this environment because graphifyy / Python is missing

## Release Notes

* **New Features**
  * Enhanced coach chat state management with revision tracking and improved summary synchronization.
  * Added WebSocket connection management for real-time coach chat updates.

* **Bug Fixes**
  * Fixed stale cache merge behavior to preserve newer summaries when older metadata updates arrive.
  * Improved handling of conflicting summary versions.

* **Refactoring**
  * Reorganized settings service architecture into modular components.
  * Refactored coach chat to delegate state management and improve code maintainability.

* **Tests**
  * Added comprehensive test coverage for cache behavior, state management, and settings operations.

[Review Change Stack](https://app.coderabbit.ai/change-stack/andrzejwitkowski/AiWattCoach/pull/267?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: frontend/src/features/coach/context/CoachSessionCache.tsx [hunk 1]
```diff
@@ -1,36 +1,85 @@
...
-    [],
-  );
-
-  const setSummary = useCallback((summary: WorkoutSummary) => {
+  const upsertFullSummary = useCallback((summary: WorkoutSummary) => {
     summariesRef.current.set(summary.workoutId, summary);
+    setRevision((current) => current + 1);
+  }, []);
+
+  const hydrateMetadataSummaries = useCallback((requestedWorkoutIds: string[], summariesForRequest: WorkoutSummary[]) => {
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply frontend/src/features/coach/context/CoachSessionCache.tsx]: **Fixed in commit 2b1af10.** Removed unused `sessionCache` import and variable from `CoachPageLayout.tsx`.

File: frontend/src/features/coach/components/CoachPageLayout.tsx [hunk 2]
```diff
@@ -42,14 +42,11 @@ export function CoachPageLayout({ apiBaseUrl }: CoachPageLayoutProps) {
...
   );
-  const cachedSummary = selectedItem ? sessionCache.getSummary(selectedItem.id) : null;
 
   const chat = useCoachChat({
     apiBaseUrl,
     workoutId: selectedItem?.id ?? null,
     aliasRange: workoutList.visibleWeekRange,
-    cachedSummary,
-    onSummaryLoaded: sessionCache.setSummary,
   });
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply frontend/src/features/coach/components/CoachPageLayout.tsx:48]: **Fixed in commit 2b1af10.** Removed unused `sessionCache` import and variable from `CoachPageLayout.tsx`.

File: frontend/src/features/coach/hooks/coachWorkoutListItems.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+      hasConversation: false,
+    });
+  }
+
+  return items.sort((left, right) => right.startDateLocal.localeCompare(left.startDateLocal));
+}
+
+export function isSameDay(left: Date, right: Date): boolean {
+  return left.getTime() === right.getTime();
+}
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply frontend/src/features/coach/hooks/coachWorkoutListItems.ts:189]: **Fixed.** Changed `isSameDay` to compare `getFullYear/getMonth/getDate` calendar components instead of millisecond timestamps. Now returns true for any two Date objects on the same calendar day regardless of time-of-day.

File: frontend/src/features/coach/hooks/useCoachChatSocket.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+export function buildWorkoutSummaryWebSocketUrl(apiBaseUrl: string, workoutId: string): string {
+  const path = `/api/workout-summaries/${workoutId}/ws`;
+
+  if (!apiBaseUrl) {
+    return `${buildProtocol(window.location.protocol)}//${window.location.host}${path}`;
+  }
+
+  if (apiBaseUrl.startsWith('/')) {
+    return `${buildProtocol(window.location.protocol)}//${window.location.host}${apiBaseUrl}${path}`;
+  }
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply frontend/src/features/coach/hooks/useCoachChatSocket.ts:37]: **Fixed.** Added `apiBaseUrl.replace(/\/+$/, "")` before concatenation to prevent double slash when apiBaseUrl ends with `/`.

File: frontend/src/features/coach/hooks/useCoachChatSocket.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+          socketRef.current = null;
+          socketWorkoutIdRef.current = null;
+          setIsConnected(false);
+        }
+        if (pendingSocketRef.current?.socket === socket) {
+          pendingSocketRef.current = null;
+        }
+        setIsCoachTyping(false);
+        clearReplyProgress();
+      });
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply frontend/src/features/coach/hooks/useCoachChatSocket.ts:188]: **Fixed.** Added `reject` field to `PendingSocketState` and call it in the `close` handler when a pending socket closes before opening. Prevents `connectSocket()` from hanging indefinitely.

File: src/adapters/mongo/settings/documents.rs [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
+pub(super) struct IntervalsDocument {
+    pub(super) api_key: Option<String>,
+    pub(super) athlete_id: Option<String>,
+    #[serde(default)]
+    pub(super) connected: bool,
+    pub(super) updated_at_epoch_seconds: Option<i64>,
+    #[serde(default)]
+    pub(super) updated_at: Option<DateTime>,
+}
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply src/adapters/mongo/settings/documents.rs]: **Fixed.** Added custom `Debug` impls for `AiAgentsDocument` and `IntervalsDocument` with `RedactedOptionalText` for API key fields, matching existing `WahooDocument`/`CyclingDocument` pattern.

File: src/adapters/mongo/settings/tests/support.rs [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        Err(error) => {
+            if std::env::var("REQUIRE_MONGO_IN_CI").as_deref() == Ok("true") {
+                panic!("mongo settings test requires Mongo in CI: {error}");
+            }
+            eprintln!(
+                "skipping mongo settings test: failed to connect to Mongo at {mongo_uri}: {error}"
+            );
+            None
+        }
+    }
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply src/adapters/mongo/settings/tests/support.rs:77]: **Fixed.** Added `redact_mongo_uri()` that replaces userinfo with `<redacted>` (e.g. `mongodb://user:pass@host` becomes `mongodb://<redacted>@host`).

File: src/domain/settings/service/tests/intervals.rs [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+
+    let stored = poll_states.stored();
+    assert!(stored
+        .iter()
+        .all(|state| state.next_due_at_epoch_seconds == i64::MAX));
+    assert!(stored.iter().all(|state| state.cursor.is_none()));
+    assert!(stored
+        .iter()
+        .all(|state| state.backoff_until_epoch_seconds.is_none()));
+    assert!(stored.iter().all(|state| state.last_error.is_none()));
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply src/domain/settings/service/tests/intervals.rs]: **Fixed.** Changed to `connected: true` so the test exercises the credential-change path while Intervals remains enabled. Updated assertion to match `now_epoch_seconds` (1_700_000_000) instead of `i64::MAX`.

File: src/domain/settings/service/tests/support.rs [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+}
+
+impl UserSettingsRepository for InMemoryUserSettingsRepository {
+    fn find_by_user_id(
+        &self,
+        _user_id: &str,
+    ) -> BoxFuture<Result<Option<UserSettings>, SettingsError>> {
+        let settings = self.settings.clone();
+        Box::pin(async move { Ok(settings.lock().unwrap().clone()) })
+    }
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply src/domain/settings/service/tests/support.rs:53]: **Fixed.** Added `.filter(|s| s.user_id == user_id)` to `find_by_user_id` and all `update_*` methods. Test doubles now catch cross-user regressions.

File: src/domain/settings/service/updates.rs [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+        let history_ftp_watts = updated.cycling.ftp_watts.map(|ftp| ftp as i32).unwrap_or(0);
+        if let Err(error) = repository
+            .upsert(FtpHistoryEntry {
+                user_id: user_id.to_string(),
+                effective_from_date: epoch_seconds_to_utc_date(now),
+                ftp_watts: history_ftp_watts,
+                source: FtpSource::Settings,
+                created_at_epoch_seconds: now,
+                updated_at_epoch_seconds: now,
+            })
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply src/domain/settings/service/updates.rs]: **Fixed.** Changed `unwrap_or(0)` to `let Some(watts) = ... else { return; }`. Skips upsert when FTP is cleared instead of creating a synthetic 0W entry.

File: src/adapters/mongo/settings/mapping.rs [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+                .map(|provider| provider.as_str().to_string()),
+            selected_model: settings.ai_agents.selected_model.clone(),
+        },
+        intervals: IntervalsDocument {
+            api_key: settings.intervals.api_key.clone(),
+            athlete_id: settings.intervals.athlete_id.clone(),
+            connected: settings.intervals.connected,
+            updated_at_epoch_seconds: None,
+            updated_at: None,
+        },
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply src/adapters/mongo/settings/mapping.rs:92]: **Deferred.** Needs investigation into adapter-owned vs domain-owned semantics before changing the mapping. Will address in a separate PR.

File: src/adapters/mongo/settings/repository.rs [hunk 12]
```diff
@@ -0,0 +1,285 @@
...
+                            "updated_at": optional_epoch_seconds_to_bson_datetime(Some(updated_at), "updated_at")
+                                .map_err(SettingsError::Repository)?,
+                        }
+                    },
+                )
+                .await
+                .map_err(|e| SettingsError::Repository(e.to_string()))?;
+            Ok(())
+        })
+    }
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply src/adapters/mongo/settings/repository.rs:284]: **Deferred.** This is a behavior change (propagating NotFound on missing user_id). Will address in a separate PR with the `ensure_update_matched` helper.

File: src/domain/settings/service/updates.rs [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+            now,
+        )
+        .await
+        {
+            tracing::warn!(
+                user_id = %user_id,
+                error = %error,
+                "interval settings were saved but provider poll state sync failed"
+            );
+        }
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply src/domain/settings/service/updates.rs:66]: **Deferred.** Behavior change — needs design decision on whether to propagate the error or create a durable retry record. Will address in a separate PR.

File: frontend/src/features/coach/context/CoachSessionCache.tsx [hunk 14]
```diff
@@ -1,36 +1,85 @@
...
+
+  if (incoming.updatedAtEpochSeconds < existing.updatedAtEpochSeconds) {
+    return existing;
+  }
+
+  if (incoming.messages.length === 0 && existing.messages.length > 0) {
+    return {
+      ...incoming,
+      messages: existing.messages,
+      hasCoachMessage: incoming.hasCoachMessage ?? existing.hasCoachMessage,
```
[the remark this answers is not in the document]
Comment (andrzejwitkowski) [reply frontend/src/features/coach/context/CoachSessionCache.tsx:21]: **Deferred.** This is about the metadata hydration merge behavior which is already covered by the `CoachSessionCache.test.tsx` regression tests added in this PR. The existing merge logic correctly preserves full summaries when metadata has empty messages. Will revisit if specific edge cases are identified.
